src/main/java/com/esri/samples/na/routing_around_barriers/README.md

```diff
@@ -0,0 +1,91 @@
+<h1>Routing around barriers</h1>
+
+<p>Find a route that reaches all stops without crossing any barriers.</p>
+
+<p><img src="RoutingAroundBarriers.png"/></p>
+
+<h2>Use case</h2>
+
+<p>You can define barriers to avoid unsafe areas, for example flooded roads, when planning the most efficient route to evacuate a hurricane zone. When solving a route, barriers allow you to define portions of the road network that cannot be traversed. You could also use this functionality to plan routes when you know an area will be inaccessible due to a community activity like an organized race or a market night.
+   
+ In some situations, it is further beneficial to find the most efficient route that reaches all stops, reordering them to reduce travel time. For example, a delivery service may target a number of drop-off addresses, specifically looking to avoid congested areas or closed roads, arranging the stops in the most time-effective order.</p>
+
+<h2>How to use the sample</h2>
+
+<p>Use the Edit Mode toggle buttons to select whether to add Stops or Barriers to the route. Click 'Determine Route' to find the route and display it. Select 'Allow stops to be re-ordered' to find the best sequence. Select 'Preserve first stop' if there is a known start point, and 'Preserve last stop' if there is a known final destination.</p>
+
+<h2>How it works</h2>
+
+<ol>
+  <li>Create the route task by calling <code>RouteTask.CreateAsync(_serviceUrl)</code> with the URL to a Network Analysis route service.</li>
+  <li>Get the default route parameters for the service by calling <code>_routeTask.CreateDefaultParametersAsync</code>.</li>
+
+  <li>When the user adds a stop, add it to the route parameters.
+    <ol>
+      <li>Normalize the geometry; otherwise the route job would fail if the user included any stops over the 180th degree meridian.</li>
+      <li>Get the name of the stop by counting the existing stops - <code>_stepsOverlay.Graphics.Count + 1</code>.</li>
+      <li>Create a composite symbol for the stop. This sample uses a pushpin marker and a text symbol.</li>
+      <li>Create the graphic from the geometry and the symbol.</li>
+      <li>Add the graphic to the stops graphics overlay.</li>
+    </ol>
+  </li>
+
+  <li>When the user adds a barrier, create a polygon barrier and add it to the route parameters.
+    <ol>
+      <li>Normalize the geometry (see <strong>3i</strong> above).</li>
```

Comment:
Isn't it more correct to say 3i, since the second-order lists use roman number?

Reply:
the HTML in your README will render to display 3.1, rather than 3.i (as Github shows it - I don't know why this occurs, different rendering of HTML in Markdown. I use MarkdownPad 2 to do README formatting checks). However, that reminds me there is a bug when rendering ordered lists within ordered lists as this README does, when pushed to our Dev samples page. Replace the nested ordered lists with unordered lists (this will lose the numbering but that's ok - can always say something like "normalize the geometry as in step 3").

Reply:
Good point. It looks like the HTML Markdown gets rendered either 3.1 or 3.i depending on the system setup! Something to keep in mind.
Did not know about the ordered list bug. Changed all the second order lists to <ol> now.